### PR TITLE
projects: Add SWIOT1L project

### PR DESCRIPTION
**Pull request by CiprianRegus**
## Pull Request Description

This is required for the release of the SWIOT1L analog.com page. Has been tested with all the other changes implemented in separate pull requests related to the ADIN1110 and AD74413R.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: projects/swiot1l/Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+include src.mk
+
+include ../../tools/scripts/generic.mk
+
+CFLAGS += -DNO_OS_LWIP_NETWORKING
+CFLAGS += -DNO_OS_STATIC_IP
+CFLAGS += -DIIO_IGNORE_BUFF_OVERRUN_ERR
+CFLAGS += -DNO_OS_LWIP_INIT_ONETIME=1
+
```
Comment (amiclaus) [inline projects/swiot1l/Makefile:16]: nit: extra line

File: projects/swiot1l/Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Select the example you want to enable by chossing y for enabling and n for disabling
+PLATFORM = maxim
+TARGET = max32650
+LIBRARIES += lwip
```
Comment (amiclaus) [inline projects/swiot1l/Makefile:4]: this goes usually in src.mk

File: projects/swiot1l/src/common/common_data.c [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+	.handle = AD74413R_GPIO_CB_HANDLE,
+};
+
+struct iio_hw_trig_init_param ad74413r_gpio_trig_ip = {
+	.irq_id = AD74413R_GPIO_TRIG_IRQ_ID,
+	.irq_trig_lvl = NO_OS_IRQ_EDGE_FALLING,
+	.cb_info = gpio_cb_info,
+	.name = AD74413R_GPIO_TRIG_NAME,
+};
+
```
Comment (amiclaus) [inline projects/swiot1l/src/common/common_data.c:315]: nit: extra line.

File: projects/swiot1l/src/platform/maxim/parameters.c [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+struct max_gpio_init_param vddioh_gpio_extra = {
+	.vssel = MXC_GPIO_VSSEL_VDDIOH,
+};
+
+struct max_i2c_init_param vddioh_i2c_extra = {
+	.vssel = MXC_GPIO_VSSEL_VDDIOH,
+};
+
```
Comment (amiclaus) [inline projects/swiot1l/src/platform/maxim/parameters.c:60]: nit: extra line

**Comment by CiprianRegus** [comment]
Addressed review comments

**Review by amiclaus — APPROVED** [review]

**Pull request merged**